src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -1,8 +1,156 @@
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfigurator;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.DigitalInput;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants;
 import frc.robot.RobotMap;
 
 public class Elevator extends SubsystemBase {
-    
+    public enum ElevatorHeight {
+        HOME,
+        SCORE_L4,
+        SCORE_L3,
+        SCORE_L2,
+        SCORE_L1,
+        SHALLOW_PREP,
+        SHALLOW_CLIMB;
+    }
+
+    private TreeMap<ElevatorHeight, Double> elevatorPositions = new TreeMap<ElevatorHeight, Double>();
+
+    private ElevatorHeight desiredPosition = ElevatorHeight.HOME;
+    DigitalInput limitSwitchHome = new DigitalInput(ElevatorCal.ELEVATOR_LIMIT_SWITCH_DIO_HOME);
+    DigitalInput limitSwitchBelowHome = new DigitalInput(ElevatorCal.ELEVATOR_LIMIT_SWITCH_DIO_BELOWHOME);
+    DigitalInput limitSwitchTop = new DigitalInput(ElevatorCal.ELEVATOR_LIMIT_SWITCH_DIO_TOP);
+    private TalonFX leftMotor = new TalonFX(RobotMap.LEFT_ELEVATOR_MOTOR_CAN_ID);
+    private TalonFX rightMotor = new TalonFX(RobotMap.RIGHT_ELEVATOR_MOTOR_CAN_ID);
+
+    private final TrapezoidProfile m_profile_scoring = new TrapezoidProfile(
+            new TrapezoidProfile.Constraints(ElevatorCal.MAX_VELOCITY_IN_PER_SECOND_SCORE,
+                    ElevatorCal.MAX_ACCELERATION_IN_PER_SECOND_SQUARED_SCORE));
+
+    private final TrapezoidProfile m_profile_climbing = new TrapezoidProfile(
+        new TrapezoidProfile.Constraints(ElevatorCal.MAX_VELOCITY_IN_PER_SECOND_CLIMB,
+                ElevatorCal.MAX_ACCELERATION_IN_PER_SECOND_SQUARED_CLIMB));
+
+    private boolean isScoring = true;
+    private int currentSlotValue = 0; // 0 is for scoring and 1 is for climbing
+
+    private TrapezoidProfile.State m_setpoint = new TrapezoidProfile.State();
+    private TrapezoidProfile.State m_goal = new TrapezoidProfile.State();
+
+    public Elevator() {
+
+        elevatorPositions.put(ElevatorHeight.HOME, ElevatorCal.POSITION_HOME_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L4, ElevatorCal.POSITION_SCORE_L4_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L3, ElevatorCal.POSITION_SCORE_L3_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L2, ElevatorCal.POSITION_SCORE_L2_INCHES);
+        elevatorPositions.put(ElevatorHeight.SCORE_L1, ElevatorCal.POSITION_SCORE_L1_INCHES);
+        elevatorPositions.put(ElevatorHeight.SHALLOW_PREP, ElevatorCal.POSITION_PRE_SHALLOWCLIMB_INCHES);
+        elevatorPositions.put(ElevatorHeight.SHALLOW_CLIMB, ElevatorCal.POSITION_SCORE_SHALLOWCLIMB_INCHES);
```

Comment:
for consistency would just rename the ElevatorCal values as "POSITION_SHALLOW_PREP_INCHES" and "POSITION_SHALLOW_CLIMB_INCHES"